lib/galaxy/datatypes/genetics.py

```diff
@@ -863,16 +863,16 @@ def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> N
         pp = os.path.join(dataset.extra_files_path, pn)
         dataset.metadata.pheno_path = pp
         try:
-            with open(pp) as f:
-                pf = f.readlines()  # read the basename.phenodata in the extra_files_path
+            with open(pp) as file:
+                pf = file.readlines()  # read the basename.phenodata in the extra_files_path
         except Exception:
             pf = None
         if pf:
-            h = pf[0].strip()
-            h = h.split("\t")  # hope is header
-            h = [escape(x) for x in h]
-            dataset.metadata.column_names = h
-            dataset.metadata.columns = len(h)
+            header = pf[0].strip()
+            columns = header.split("\t")  # hope is header
+            columns = [escape(x) for x in columns]
```

Comment:
You can combine them in a single statement:
```suggestion
            columns = [escape(x) for x in header.split("\t")]  # hope is header
```